functions/onprem/orborus/orborus.go

```diff
@@ -666,129 +662,210 @@ func handleBackendImageDownload(ctx context.Context, images string) error {
-			kubernetesImage = image
+		if len(foundNamespace) > 0 {
+			kubernetesNamespace = foundNamespace
+			os.Setenv("KUBERNETES_NAMESPACE", kubernetesNamespace)
 		}
-		log.Printf("[DEBUG] Using Kubernetes worker image '%s'", kubernetesImage)
-		// image = "shuffle-worker:v1" //hard coded image name to test locally
+	}
 
-		envMap := make(map[string]string)
-		for _, envStr := range env {
-			parts := strings.SplitN(envStr, "=", 2)
-			if len(parts) == 2 {
-				envMap[parts[0]] = parts[1]
-			}
+	if len(kubernetesNamespace) == 0 {
+		kubernetesNamespace = "default"
+	}
+
+	kubernetesImage := os.Getenv("SHUFFLE_KUBERNETES_WORKER")
+	if len(kubernetesImage) == 0 {
+		kubernetesImage = image
+	}
+	log.Printf("[DEBUG] Using Kubernetes worker image '%s'", kubernetesImage)
+	// image = "shuffle-worker:v1" //hard coded image name to test locally
+
+	envMap := make(map[string]string)
+	for _, envStr := range env {
+		parts := strings.SplitN(envStr, "=", 2)
+		if len(parts) == 2 {
+			envMap[parts[0]] = parts[1]
 		}
+	}
 
-		// While testing:
-		// kubectl delete pods --all --all-namespaces; kubectl delete services --all --all-namespaces
-		pod := &corev1.Pod{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:   identifier,
-				Labels: map[string]string{"app": "shuffle-worker"},
+	containerLabels := map[string]string{
+		"container": "shuffle-worker",
+	}
+
+	containerAttachment := corev1.Container{
+		Name:  identifier,
+		Image: kubernetesImage,
+		Env:   buildEnvVars(envMap),
+		
+		//ImagePullPolicy: "Never",
+		ImagePullPolicy: corev1.PullIfNotPresent,
+	}
+
+	podname := shuffle.GetPodName()
+
+	ctx := context.Background()
+
+	if len(podname) > 0 {
+		_, err := shuffle.GetCurrentPodNetworkConfig(ctx, clientset, kubernetesNamespace, podname)
+		if err != nil {
+			log.Printf("[ERROR] Failed getting current pod network: %s", err)
+		} else {
+			log.Printf("[DEBUG] Current pod found!")
+			// currentPodStatus = k8s.io/api/core/v1.PodStatus
+		}
+	}
+
+
+	// While testing:
+	// kubectl delete pods --all --all-namespaces; kubectl delete services --all --all-namespaces
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   identifier,
+			Labels: containerLabels,
+		},
+		Spec: corev1.PodSpec{
+			RestartPolicy: "Never",
+			// DNSPolicy:     "Default",
+			DNSPolicy: 	corev1.DNSClusterFirst,
+			// NodeSelector: map[string]string{
+			// 	"node": "master",
+			// },
+			Containers: []corev1.Container{
+				containerAttachment,
 			},
-			Spec: corev1.PodSpec{
-				RestartPolicy: "Never",
-				DNSPolicy:     "Default",
-				// NodeSelector: map[string]string{
-				// 	"node": "master",
-				// },
-				Containers: []corev1.Container{
-					{
-						Name:  identifier,
-						Image: kubernetesImage,
-						Env:   buildEnvVars(envMap),
-						
-						//ImagePullPolicy: "Never",
-						ImagePullPolicy: corev1.PullIfNotPresent,
-						//ImagePullPolicy: "Always",
-					},
+		},
+	}
+
+	// Check if running on ARM or x86 to download the correct image
```

Comment:
You're saying this is checking, but is it really? I don't see you actually going through the list anywhere. Please remove/comment if unused

Reply:
huh, seems like i wanted to do something and then decided there wasn't a need